Sources/TuistCore/Utils/OtoolController.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import RxSwift
+import TSCBasic
+import TuistSupport
+
+enum OtoolErrors: FatalError, Equatable {
+    case invalidOutput
+
+    var description: String {
+        switch self {
+        case .invalidOutput: return "The output is not in the format tuist expects"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .invalidOutput: return .abort
+        }
+    }
+}
+
+/// Otool controller protocol passed in initializers for dependency injection
+public protocol OtoolControlling {
+    /// Returns the paths to dlyb dependencies.
+    /// - Parameter path: Path to the given dependency
+    /// - Returns Array of dependencies returned by the command.
+    func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]>
+}
+
+/// OtoolController
+/// Used to find out against which dynamic libraries a certain binary is linked.
+public struct OtoolController: OtoolControlling {
+    private let system: Systeming
+
+    public init(system: Systeming = System.shared) {
+        self.system = system
+    }
+
+    public func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]> {
+        let arguments = ["otool", "-L", path.pathString]
+
+        return System.shared.observable(arguments)
+            .mapToString()
+            .collectAndMergeOutput()
+            .map { try OtoolOutput(filePath: path, output: $0) }
+            .map(\.paths)
+            .asSingle()
```

Comment:
I can see that you had to add RxSwift and RxBlocking to TuistCore's dependencies. Im thinking if we can avoid it. 🤔

Reply:
This is the result of this [suggestion](https://github.com/tuist/tuist/pull/2025#discussion_r524501090). 
Probably right now is not that useful, but I guess it will in the future.

Reply:
I think it's fine. What's important though is to only do blocking at the service level, where all the combination of asynchronous events take place. Utilities like this one that can run in parallel should be implemented as asynchronous events as you just did.